src/CommunityToolkit.Aspire.Hosting.OpenTelemetryCollector/EnvironmentVariableHook.cs

```diff
@@ -0,0 +1,59 @@
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Hooks to add the OTLP environment variables to the various containers
+/// </summary>
+/// <param name="logger"></param>
+public class EnvironmentVariableHook(ILogger<EnvironmentVariableHook> logger) : IDistributedApplicationLifecycleHook
```

Comment:
On top of moving this implementation to the eventing model, we should also use this [OtlpExporterAnnotation](https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.applicationmodel.otlpexporterannotation?view=dotnet-aspire-9.0) to detect which resources should be impacted.